Defining the manual process for approving PRs (#23)
RedHatInsights/inventory-schemas: README.md

Document version is reflected in the filename (_v1.yaml_) and under _$version_ root key. _$id_ is a file name used in
the Host Inventory.

## Process For Merging Schema Changes ##

1. Once a pull request is opened to update a pre-existing field and all the requested changes are resolved, someone from [RedHatInsights/host-based-inventory-comitters team](https://github.com/orgs/RedHatInsights/teams/host-based-inventory-committers) should open a merge request with [App-SRE](https://gitlab.cee.redhat.com/service/app-interface/-/tree/master/data/services/insights/host-inventory/queries) to query the database for potential impact to the HBI reporters.
   

> 2. The person from HBI team will analyze the data returned from the App-SRE job and produce a report similiar to to the following and post it in the pull request thread.

Reviewer: ```suggestion
2. The HBI team member will analyze the data returned from the App-SRE job, produce a report similar to to the following, and post it in the pull request thread.
```
Change run-on sentence and fix typo

    | Reporters | Result | Count |
    | --- | --- | --- |
    | ingress | Pass | 10 |
    | ingress | Fail | 2 |
    | puptoo | Pass | 37 |
    | puptoo | Fail | 42 |

> 3. If the report seems satisfactory and if there are no more concerns, someone from the HBI team will merge the PR.

Reviewer: ```suggestion
3. If the report seems satisfactory and there are no more concerns, someone from the HBI team will merge the PR.
```
I'm actually not sure about this one, but it sounds more natural to me? You can leave it in if you disagree :)

> 4. However, in the event that the report is unsatisfactory, e.g. a high number of failures from one or multiple reporters, coordination must happen between the person who opened the PR, the HBI team and the stakeholder from the reporter(s) with the failures. During the conversation a decision must be made whether to change the PR or to update the reporter with the failures.

Reviewer: ```suggestion
4. In the event that the report is unsatisfactory (e.g. it shows a high number of failures from one or multiple reporters), the PR owner must coordinate with the HBI team and the stakeholders from the reporter(s) that show failures. Together, they must decide whether to change the PR or to update the reporter(s).
```
I changed some things around to remove passive voice and (hopefully) improve readability

> 5. If the pull request was opened to add a new field rather than update a pre-existing one, no report like the one above needs to be generated and someone from the HBI team can merge it if there are no concerns.

Reviewer: ```suggestion
5. If the pull request was opened to add a new field rather than update a pre-existing one, no report like the one above needs to be generated; someone from the HBI team can merge it if there are no concerns.
```
I divided up the thoughts with a semicolon because it was running on a bit. If you prefer, you can use a period and separate it into two sentences instead.

> 6. Once a PR to update the system profile schema is merged, the representative from the HBI team who merged the PR can go ahead and open a PR in the HBI repository to reflect those changes if no such PR has already been opened in the HBI repository.

Reviewer: ```suggestion
6. Once the PR is merged, the HBI team member who merged it can open another PR in the HBI repository to reflect those changes.
```
Removes some redundancy for clarity's sake. I'm also suggesting we remove that final clause because I feel it's implied... what do you think?

### Stakeholders ###

* RHSM - [Kevin Howell](https://github.com/kahowell)
* Yupana - [Chris Duryee](https://github.com/beav)
* Puptoo - [Stephen Adams](https://github.com/stevehnh)
* BU - [Jerome Marc](https://github.com/jeromemarc)

## To do ##

> - [ ] Write more to do items

Reviewer: LOL

   
2. The HBI team member will analyze the data returned from the App-SRE job, produce a report similar to to the following, and post it in the pull request thread.

    | Reporters | Result | Count |
    | --- | --- | --- |
    | ingress | Pass | 10 |
    | ingress | Fail | 2 |
    | puptoo | Pass | 37 |
    | puptoo | Fail | 42 |

3. If the report seems satisfactory and there are no more concerns, someone from the HBI team will merge the PR.

4. In the event that the report is unsatisfactory (e.g. it shows a high number of failures from one or multiple reporters), the PR owner must coordinate with the HBI team and the stakeholders from the reporter(s) that show failures. Together, they must decide whether to change the PR or to update the reporter(s).

> 5. If the pull request was opened to add a new field rather than update a pre-existing one, no report like the one above needs to be generated. Someone from the HBI team can merge it if there are no concerns.

Reviewer 2: @tahmidefaz What do you think of something like this:
If the pull request only adds new fields there is no need to generate a report since the database will not contain values to compare against for the new fields.  A repository maintainer will help in getting consensus with the schema [Stakeholders](#stakeholders) before merging the PR.

If you want we could also link the "repository maintainer" to the github team.

Author: Yes, sounds good. I will update it. Thanks! :+1:
